src/main/java/org/ihiw/management/service/UserService.java

```diff
@@ -398,6 +403,7 @@ public void changePassword(String currentClearTextPassword, String newPassword)
         	userIds.add(ihiwUser.getId());
         }
        	List<Long> projectIds = new ArrayList<Long>();
+       	projectIds.add(new Long(-1)); // Default Value, because findByUsersAndProjects crashes with an empty projectIDs list(Which is possible in some cases)
```

Comment:
...and there aren't any options to make findByUsersAndProjects more resilient? Is there maybe an additional modifier like findByUsersAndProjectsWithAcceptEmptyList or something?

Besides from that... I'm always pro cost-efficient solutions :-D

Reply:
https://rzymek.github.io/post/jpa-empty-in/ discusses the issue, passing an empty list as a parameter in the sql query ends up with some syntax errors.

Reply:
In the interest of cost-efficient but logical solutions, I added a check: If projectIds is Empty, then we use
findParentlessByCreatedByIn

otherwise use
findParentlessByUsersAndProjects

Slightly more elegant than putting a placeholder int in the list.

This should only come up if a user is a "PROJECT_LEADER" role, but is not assigned as the leader of any specific project. In this case they can still see labmembers' uploads, (but no extras from their project leader role)